Mathlib/FieldTheory/CardinalEmb.lean

```diff
@@ -0,0 +1,344 @@
+/-
+Copyright (c) 2024 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.FieldTheory.SeparableClosure
+import Mathlib.FieldTheory.PurelyInseparable
+import Mathlib.LinearAlgebra.FreeAlgebra
+import Mathlib.Order.Interval.Set.WithBotTop
+import Mathlib.Order.DirectedInverseSystem
+
+/-!
+# Number of embeddings of an infinite algebraic field extension into the algebraic closure
+
+We show that if `E/F` is an infinite-dimensional algebraic extension, then
+`#(Field.Emb F E) = 2 ^ Field.sepDegree F E`. If `E/F` is moreover separable, then
```

Comment:
According to the main result, this holds only if `Field.sepDegree F E` is infinite. So you should say "an algebraic extension with infinite separable degree".